packages/fuels-programs/src/call_utils.rs

```diff
@@ -358,42 +352,22 @@ pub(crate) fn get_single_call_instructions(
         None => instructions.push(op::call(0x10, 0x11, 0x12, RegId::CGAS)),
     };
 
-    // The instructions are different if you want to return data that was on the heap
-    if let Some(inner_type_byte_size) = output_param_type.heap_inner_element_size()? {
-        instructions.extend([
-            // The RET register contains the pointer address of the `CALL` return (a stack
-            // address).
-            // The RETL register contains the length of the `CALL` return (=24 because the
-            // Vec/Bytes/String struct takes 3 WORDs).
-            // We don't actually need it unless the Vec/Bytes/String struct encoding changes in the
-            // compiler.
-            // Load the word located at the address contained in RET, it's a word that
-            // translates to a heap address. 0x15 is a free register.
-            op::lw(0x15, RegId::RET, 0),
-            // We know a Vec/Bytes/String struct has its third WORD contain the length of the
-            // underlying vector, so use a 2 offset to store the length in 0x16, which is a free
-            // register.
-            op::lw(0x16, RegId::RET, 2),
-            // The in-memory size of the type is (in-memory size of the inner type) * length
-            op::muli(0x16, 0x16, inner_type_byte_size as u16),
-            op::retd(0x15, 0x16),
-        ]);
-    }
+    instructions.extend(extract_heap_data(output_param_type)?);
```

Comment:
I think those comments are essential to understand the bytecode mumbo-jumbo needed there.

Reply:
Code was changed in a merge, this no longer stands as those comments were moved somewhere else.